web/src/main/webapp/heatMapArray/view.ts

```diff
@@ -0,0 +1,288 @@
+import {RemoteObjectView, CallbackCollector} from "../rpc";
+import {Dialog} from "../dialog";
+import {RangeInfo, TableView, Schema, IColumnDescription, isNumeric, isCategorical, DistinctStrings} from "../table";
+import {BasicColStats} from "../histogramBase";
+import {FullPage, Size, Resolution, IHtmlElement, ScrollBar, IScrollTarget} from "../ui";
+import {Pair, Triple} from "../util";
+import {ColumnAndRange} from "../histogramBase";
+import {HeatMapData} from "../heatmap";
+import {HeatMapArrayArgs, HeatMapArrayData} from "./types";
+import d3 = require('d3');
+
+class ColorMap {
+    public logScale: boolean;
+    constructor(public max) {
+        this.logScale = max > 20;
+    }
+
+    public setLogScale(logScale: boolean) {
+        this.logScale = logScale;
+    }
+
+    public apply(x) {
+        if (this.logScale)
+            return this.applyLog(x);
+        else
+            return this.applyLinear(x);
+    }
+
+    private applyLinear(x) {
+        return this.map(x / this.max);
+    }
+
+    private applyLog(x) {
+        return this.map(Math.log(x) / Math.log(this.max));
+    }
+
+    private map(x: number) {
+        return d3.interpolateWarm(x);
+    }
+}
+
+class CompactHeatMapView implements IHtmlElement {
+    // Size of a single heatmap in pixels
+    public static readonly size: Size = {
```

Comment:
I am worried about this being hardwired.
First of all, maybe you can reuse the Resolution class.
Second, you should try to compute the size; e.g., if there are only 3 categories, you should make each heat-map 1/3 of the page width instead of hardwiring the size here.